### PR TITLE
handles the case of running a model where a provider doesn't have A&E data

### LINKED ISSUE #648: model fails to run if a provider doesn't have data for A&E
the model tries to download data, but if a provider doesn't have an A&E department it will crash trying to download files that don't exist.

also, the model always tries to run for IP/OP/A&E, so if we fix the issue above it will still fail to run.

### PR DESCRIPTION
**Pull request by tomjemmett**
fixes #648

- **adds step to skip downloading data if it doesn't exist**
- **changes type of the data argument to model instances to be Data, not a callable**
- **adds method to test that files exist for a specified model type**
- **skips running a model type if data doesn't exist**
- **handles the edge case in sdec patching for providers without A&E data**

### CODE REVIEW

File: src/nhp/model/run.py [hunk 1]
```diff
@@ -161,6 +164,7 @@ def run_all(
                 aggregation_columns,
             )
             for m in model_types
+            if nhp_data.data_exists_for_model_type(m)
         ]
```
[the remark this answers is not in the document]
Comment (tomjemmett) [reply src/nhp/model/run.py:168]: I think this is fine to leave for now - the model shouldn't be available to providers that don't have any data in the first place

File: src/nhp/model/run.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ def noop_progress_callback(_: Any) -> Callable[[Any], None]:
 
 def run_all(
     params: dict,
-    nhp_data: Callable[[int, str], Data],
+    nhp_data: Data | Callable[[int, str], Data],
     progress_callback: Callable[[Any], Callable[[Any], None]] = noop_progress_callback,
     save_full_model_results: bool = False,
```
[the remark this answers is not in the document]
Comment (tomjemmett) [reply src/nhp/model/run.py:121]: I think this is still fine

**Comment by yiwen-h** [comment]
@tomjemmett  is there an existing model run with the provider we know has no A&E using this PR container? (preferably using sample params which has the SDEC mitigators turned on)

**Comment by tomjemmett** [comment]
no I've not run anything other than local testing

**Comment by yiwen-h** [comment]
I'll set one off as part of the PR review then

**Comment by yiwen-h** [comment]
@tomjemmett  it's run successfully and I can see results in `aggregated-model-results/dev/RRJ/test-pr649/20260727_100008/`. However I can't view the results in the outputs selection app and the record in ATS looks a bit strange

**Comment by tomjemmett** [comment]
working now - issue was with the results json

**Review by yiwen-h — APPROVED** [review]
Have tested the PR containers both with and without full results on Azure Container Instances. Thank you for sorting this!
